### PR TITLE
[Fix] MaskFormer Loss Computation Bugs

### PR DESCRIPTION
**Pull request by Bobholamovic**
### PR types
Bug fixes

### PR changes
Models

### Description

修复MaskFormer训练bug，详见注释。

### CODE REVIEW

File: paddleseg/models/losses/maskformer_loss.py [hunk 1]
```diff
@@ -345,6 +345,8 @@ def loss_masks(self, outputs, targets, indices, num_masks):
         tgt_idx = self._get_tgt_permutation_idx(indices_cpt)
         src_masks = outputs["pred_masks"]
         src_masks = src_masks[src_idx]
+        if src_masks.ndim == 2:
```
Comment (Bobholamovic) [inline paddleseg/models/losses/maskformer_loss.py:348]: 原先未考虑`src_idx`列表中只有一个有效target的情况，在该情况下会出现维度错误。

File: paddleseg/models/losses/maskformer_loss.py [hunk 2]
```diff
@@ -403,12 +405,15 @@ def forward(self, logits, targets):
 
         targets = []
         for item in targets_cpt:
-            paddle.cast(item['masks'], 'bool')
-            start_idx = paddle.nonzero(
-                paddle.cast(item["labels"] == self.ignore_index, 'int64'))[
-                    0].numpy()[0]
-            index = paddle.cast(paddle.to_tensor([i for i in range(start_idx)]),
-                    'int64')
+            item['masks'] = paddle.cast(item['masks'], 'bool')
```
Comment (Bobholamovic) [inline paddleseg/models/losses/maskformer_loss.py:408]: 因`paddle.cast`非in-place操作，原先的写法未起到预期作用。

File: paddleseg/models/losses/maskformer_loss.py [hunk 3]
```diff
@@ -403,12 +405,15 @@ def forward(self, logits, targets):
...
+            invalid_indices = paddle.nonzero(
```
Comment (Bobholamovic) [inline paddleseg/models/losses/maskformer_loss.py:409]: 原先未考虑样本中不存在`ignore_index`的情况。

**Review by juncaipeng — APPROVED** [review]

**Review by shiyutang — APPROVED** [review]
LGTM
